akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -100,8 +100,14 @@ import akka.util.OptionVal
           }
           context.stop(self)
         case f: FailedBehavior ⇒
+          f.cause match {
+            case e: InterpretBehaviorException[T] ⇒
+              behavior = Behavior.canonicalize(e.failed, behavior, ctx)
+            // could throw e and handle differently in supervisorStrategy
```

Comment:
If it helps we could do these differently and modify supervisorStrategy. Everything calls Stop currently in it.

Reply:
I think it is less worse to duct tape this in into supervision than it is to put it in the actor adapter.